crates/next-core/src/app_source.rs

```diff
@@ -347,11 +347,11 @@ pub async fn create_app_source(
         output_path,
     );
 
+    let injected_env = env_for_js(EnvMapVc::empty().into(), false, next_config);
+    let env = CustomProcessEnvVc::new(env, next_config.env()).as_process_env();
```

Comment:
Why is that needed? shouldn't next config env already be applied to `env`?

Reply:
No, they're applied in `env_for_js` and I wanted the unescaped variables for the real env variables

Reply:
I think we could reorder a few things, then?
```diff
-    let injected_env = env_for_js(EnvMapVc::empty().into(), false, next_config);
+    let env = CustomProcessEnvVc::new(env, next_config.env()).as_process_env();       
+    let injected_env = env_for_js(env, false, next_config);
```

And remove the `next_config.env()` part from `env_for_js`?